lib/drab/browser.ex

```diff
@@ -296,4 +297,161 @@ defmodule Drab.Browser do
     window.history.pushState({}, "", #{Drab.Core.encode_js(url)});
     """)
   end
+
+  @doc """
+  Synchronously set a cookie on the client side
+  
+  Returns tuple {status, return_value}, where status could be :ok, :error or :timeout, and return value contains the output computed by the Javascript or the error message.  
+
+  ### Parameters:
+
+  * `socket`  - the Drab socket
+  * `key`     - the cookie name
+  * `value`   - the cookie value, can be an Elixir data structure
+        
+        
+  ### Options:
+
+  * `max-age` :
+   
+       max-age > 0    - sets the cookie max-age, in seconds
+     
+       max-age == 0   - sets a session cookie
+     
+       max-age < 0    - delete the cookie
+     
+     
+  * `path` - the cookie path, default `"/"`
+  
+  * `encode` - encodes the cookie in Base64. The encoding is needed if you want to store values with comma, semicolon, quotes, etc or Elixir data structures.   
+  
+  
+  ### Example:
+  
+        iex> Drab.Browser.set_cookie(socket, "Items", [%{id: 001, name: "foo"}, %{id: 002, name: "bar"}], max_age: 3*24*60*60, encode: true)
+        {:ok, result}
+  """
+  # @spec exec_js(Phoenix.Socket.t(), String.t(), Keyword.t()) :: result
+  def set_cookie(socket, key, value, options \\ []) do
+    # Options
+    max_age = Keyword.get(options, :max_age, 0)
+    path = Keyword.get(options, :path, "/")
+
+    # Prepare
+    encoded_value = encode_value(value, options)
+    expires = cookie_expires(socket, max_age)
+
+    # Set cookie
+    exec_js(socket, "document.cookie='#{key}=#{encoded_value}; expires=#{expires}; path=#{path};'")
+  end
+
+  @doc """
+  Exception raising version of `set_cookie/4`
+  """
+  def set_cookie!(socket, key, value, options \\ []) do
+    # Options
+    max_age = Keyword.get(options, :max_age, 0)
+    path = Keyword.get(options, :path, "/")
+
+    # Prepare
+    encoded_value = encode_value(value, options)
+    expires = cookie_expires(socket, max_age)
+
+    # Set cookie
+    exec_js!(socket, "document.cookie='#{key}=#{encoded_value}; expires=#{expires}; path=#{path};'")
+  end
+
+  @doc """
+  Delete a cookie.
+  
+  ### Parameters:
+
+  * `socket`  - the Drab socket
+  * `key`     - the cookie name
+
+  ### Example:
+  
+        iex> Drab.Browser.delete_cookie(socket, "Items")
+        {:ok, result}
+  """
+  def delete_cookie(socket, key) do
+    set_cookie(socket, key, "", max_age: -1)
+  end
+
+  @doc """
+  Exception raising version of `delete_cookie/2`
+  """
+  def delete_cookie!(socket, key) do
+    set_cookie!(socket, key, "", max_age: -1)
+  end
+
+  @doc """
+  Retrieve all cookies from browser.
+
+  ### Parameters
+
+  * `socket` - The Drab socket
+
+  ### Example:
+  
+        iex> Drab.Browser.cookies(socket)
+        {:ok, result}
+  """
+  def cookies(socket) do
+    exec_js(socket, "document.cookie")
```

Comment:
It would be better to extract cookies to some nice Elixir structure, like map, where the cookie name is a key and the value is the cookie value.

Reply:
Yes. I'll rename the original as `raw_cookies/1` and I'll write the new code under `cookies/1`

Reply:
Please try to take a look to the last PR to see if this is what you expected